lib/ex_webrtc/peer_connection/configuration.ex

```diff
@@ -12,32 +66,157 @@ defmodule ExWebRTC.PeerConnection.Configuration do
+  * `audio_codecs` - list of audio codecs to use.
+  Use `default_audio_codecs/0` to get a list of default audio codecs.
+  This option overrides default audio codecs.
+  If you wish to add codecs to default ones do 
+  `audio_codecs: Configuration.default_audio_codecs() ++ my_codecs`
+  * `video_codecs` - the same as `audio_codecs` but for video.
+  * `rtp_hdr_extensions` - list of RTP header extensions to use.
+  MID extension is enabled by default and cannot be turned off.
+  If an extension can be used both for audio and video media, it
+  will be added to every mline.
+  If an extension is audio-only, it will only be added to audio mlines.
+  If an extension is video-only, it will only be added to video mlines.
+
+  Besides options listed above, ExWebRTC uses the following config:
   * bundle_policy - max_bundle
   * ice_candidate_pool_size - 0
   * ice_transport_policy - all
   * rtcp_mux_policy - require
 
   This config cannot be changed.
   """
-  @type options() :: [ice_servers: [ice_server()]]
+  @type options() :: [
+          ice_servers: [ice_server()],
+          audio_codecs: [RTPCodecParameters.t()],
+          video_codecs: [RTPCodecParameters.t()],
+          rtp_hdr_extensions: [rtp_hdr_extension()]
+        ]
 
   @typedoc false
-  @type t() :: %__MODULE__{ice_servers: [ice_server()]}
+  @type t() :: %__MODULE__{
+          ice_servers: [ice_server()],
+          audio_codecs: [RTPCodecParameters.t()],
+          video_codecs: [RTPCodecParameters.t()],
+          audio_rtp_hdr_exts: [Extmap.t()],
+          video_rtp_hdr_exts: [Extmap.t()]
+        }
+
+  defstruct ice_servers: [],
+            audio_codecs: @default_audio_codecs,
+            video_codecs: @default_video_codecs,
+            audio_rtp_hdr_exts: @mandatory_audio_rtp_hdr_exts,
+            video_rtp_hdr_exts: @mandatory_video_rtp_hdr_exts
+
+  @doc """
+  Returns a list of default audio codecs.
+  """
+  @spec default_audio_codecs() :: [RTPCodecParameters.t()]
+  def default_audio_codecs(), do: @default_audio_codecs
 
-  defstruct ice_servers: []
+  @doc """
+  Returns a list of default video codecs.
+  """
+  @spec default_video_codecs() :: [RTPCodecParameters.t()]
+  def default_video_codecs(), do: @default_video_codecs
 
   @doc false
   @spec from_options!(options()) :: t()
   def from_options!(options) do
-    config = struct!(__MODULE__, options)
+    options =
+      options
+      |> resolve_rtp_hdr_extensions()
+      |> add_mandatory_rtp_hdr_extensions()
+      # ATM, ExICE does not support relay via TURN
+      |> reject_turn_servers()
+
+    struct!(__MODULE__, options)
+  end
+
+  @doc false
+  @spec is_supported_codec(t(), RTPCodecParameters.t()) :: boolean()
+  def is_supported_codec(config, codec) do
+    # This function doesn't check if rtcp-fb is supported.
+    # Instead, `is_supported_rtcp_fb` has to be used to filter out
+    # rtcp-fb that are not supported.
+    Enum.find(
```

Comment:
`Enum.find` doesn't return a `boolean`, contrary to what the function spec suggests.